### PR TITLE
Emscripten fixes

### PR DESCRIPTION
**Pull request by guusw**
Update toolchain and various fixes for the formabble side

### CODE REVIEW

File: deps/CMakeLists.txt [hunk 1]
```diff
@@ -173,12 +173,12 @@ else()
 
   # Use builtin SDL2 port
   target_compile_options(SDL2 INTERFACE
-    "SHELL:-s USE_SDL=2"
+    "-sUSE_SDL=2"
```
Comment (sinkingsugar) [inline deps/CMakeLists.txt]: remove, we don't use SDL

File: shards/core/coro.cpp [hunk 2]
```diff
@@ -138,7 +168,7 @@ NO_INLINE void Fiber::resume() {
 }
 
 NO_INLINE void Fiber::suspend() {
-  SHLOG_TRACE("EM FIBER SWAP SUSPEND {}", reinterpret_cast<uintptr_t>(&em_fiber));
+  // SHLOG_TRACE("EM FIBER SWAP SUSPEND {}", reinterpret_cast<uintptr_t>(&em_fiber));
```
Comment (sinkingsugar) [inline shards/core/coro.cpp]: remove or re-enable?

File: shards/core/async.hpp [hunk 3]
```diff
@@ -161,6 +174,17 @@ struct TidePool {
         worker._thread.join();
     }
   }
+#else // Dummy implementation
+  void schedule(Work *work) {
+    // SHLOG_INFO("TidePool: spawning thread");
```
Comment (guusw) [inline shards/core/async.hpp]: Cleanup

File: shards/gfx/context.cpp [hunk 4]
```diff
@@ -444,7 +465,7 @@ void Context::poll(bool wait) {
     wgpuDevicePoll(wgpuDevice, false, nullptr);
   }
 #else
-  emscripten_sleep(0);
+  // emscripten_sleep(0);
```
Comment (sinkingsugar) [inline shards/gfx/context.cpp]: remove?

File: shards/input/emscripten_browser_clipboard.h [hunk 5]
```diff
@@ -0,0 +1,124 @@
+/* https://github.com/Armchair-Software/emscripten-browser-clipboard
+
+MIT License
+
+Copyright (c) 2023 Armchair-Software
```
Comment (guusw) [inline shards/input/emscripten_browser_clipboard.h]: Not used aynmore

File: shards/lang/src/eval.rs [hunk 6]
```diff
@@ -59,7 +59,7 @@ pub(crate) const EVAL_STACK_SIZE: usize = 2 * 1024 * 1024;
 #[cfg(debug_assertions)]
 const MIN_STACK_SIZE: i64 = 2 * 1024 * 1024; // 2 MB for debug builds
 #[cfg(not(debug_assertions))]
-const MIN_STACK_SIZE: i64 = 64 * 1024; // 64 KB for release builds
+const MIN_STACK_SIZE: i64 = 1 * 1024 * 1024; // 64 KB for release builds
```
Comment (sinkingsugar) [inline shards/lang/src/eval.rs]: revert

File: shards/modules/gfx/gfx.cpp [hunk 7]
```diff
@@ -22,7 +22,7 @@ namespace gfx {
 
 Context &GraphicsContext::getContext() { return *context.get(); }
 Window &GraphicsContext::getWindow() { return *window.get(); }
-SDL_Window *GraphicsContext::getSdlWindow() { return getWindow().window; }
+// SDL_Window *GraphicsContext::getSdlWindow() { return getWindow().window; }
```
Comment (guusw) [inline shards/modules/gfx/gfx.cpp]: Clean this up

File: shards/modules/sqlite/sqlite.cpp [hunk 8]
```diff
@@ -478,43 +498,40 @@ struct Transaction : public Base {
...
+      if (rc != SQLITE_OK) {
+        throw ActivationError(sqlite3_errmsg(_connection->get()));
+      }
+    }
 
     SHVar output{};
     auto state = _queries.activate(context, input, output);
 
-    await(
-        context,
```
Comment (sinkingsugar) [inline shards/modules/sqlite/sqlite.cpp:498]: we need await here

File: shards/modules/sqlite/sqlite.cpp [hunk 9]
```diff
@@ -478,43 +498,40 @@ struct Transaction : public Base {
       SH_SUSPEND(context, 0);
     }
 
-    await(
```
Comment (sinkingsugar) [inline shards/modules/sqlite/sqlite.cpp:481]: need the awaits

**Review by sinkingsugar — APPROVED** [review]

**Pull request merged**